### PR TITLE
Removes stupid shit from wonderful wardrobe.

### LINKED ISSUE #18442: wonderful wardrobe has no sanity checks
#### Description of issue
that dang thing has vamp masks in it and shit

it was traitor

(gamemode was autotraitor, vox became vamp via the mask)

### PR DESCRIPTION
**Pull request by Allspice**
Fixes #18442, and ``piss off" with that there horrendous opinion sonix.
I guess SOMEONE should get around to making a fixed list for the closet eventually, but this is a quick fix for the dumber aspects I'm aware of.

### CODE REVIEW

File: code/game/objects/items/trader.dm [hunk 1]
```diff
@@ -62,6 +62,7 @@
 /obj/structure/closet/secure_closet/wonderful/New()
 	..()
 	var/random_clothes = clothing.Copy()
+	random_clothes = random_clothes - typesof(/obj/item/clothing/mask/stone) - typesof(/obj/item/clothing/mask/morphing) - typesof(/obj/item/clothing/accessory/holomap_chip) - typesof(/obj/item/clothing/suit/space/time) - typesof(/obj/item/clothing/head/helmet/space/time) - typesof(/obj/item/clothing/gloves/warping_claws)
```
Comment (MadmanMartian) [inline code/game/objects/items/trader.dm:65]: Triple check this works

**Comment by Kurfursten** [comment]
Like I said elsewhere, I don't think there's a substantive reason to have a separate list than Spessmart. I tend to agree with the idea that you shouldn't be able to get free antag rolls this way, but I think they should be taken off the Spessmart master list entirely.

As for anything besides antag rolls, though, I'm unconvinced. Like I pointed out in Discord, the time suit is useless without the hat so it's really only for looks unless you somehow get both, which is something like a 0.02% or 1 in 5,000 chance. Some items are gonna be useless like grey jumpsuits and plasmaman scientist suits, normally the trader gets 1-2 goodies someone would actually consider buying. That's just where I stand on it, I don't think either of us is going to convince the other. Morphing masks are popular and fun. People like to buy them, and I do see them sold to crew a lot.

File: code/game/objects/items/trader.dm (hunk 1 above)
Comment (Allspice) [reply code/game/objects/items/trader.dm]: I spawned 400 closets and sdql turned up no stone masks or warping claws in the closets (but there were some of those in nullspace presumably due to the Copy()).

**Comment by Reviire** [comment]
I don't think anything should be removed from the list, if the Vox use their role as a free antag, maybe we should just give better players a chance at the role, instead? Otherwise, I don't see it as any worse than xenoarch.

File: code/game/objects/items/trader.dm (hunk 1 above)
Comment (MadmanMartian) [reply code/game/objects/items/trader.dm:65]: Odd. Why would it be spawing them in the first place. Thank you for testing

Comment (MadmanMartian) [reply code/game/objects/items/trader.dm:65]: (No clue how to approve changes via mobile) EDIT : nvm found it

**Comment by ShiftyRail** [comment]
Xenoarch requires more effort to find the meme masks than opening a locker.
It's also designed for it. Unlike SpessMart who's just one of the many `typesof` oversights that plague our codebase (and ss13 in general)

**Comment by Allspice** [comment]
Just a reminder reviire, #13616
> I've been observing it for the last 11 days and I'm yet to see a trader sell their bag of holding once. Yet I see it almost every round there's a trader on their back.
Bad theory. You can't give players something they'd find more valuable with the intention of them selling it.

That sounds like solid reasoning upon which to base additions to a role named "trader"

**Comment by Kevak** [comment]
I do not see a reason for this.

**Comment by Accipitermaris** [comment]
@Kurfursten 
From the original PR [#17484]:

>Wonderful Wardrobe can end up getting stuff like thermals, vampire cape, security plasmaman suit, cat morphing mask, medal of captaincy, advanced hardsuit helmet, security sunglasses... or even just 19 random worthless jumpsuits and a maid outfit... It uses the completely unaltered list from Spessmart. **This might need more blacklisting or less items, I'll let github help me decide on that.**

**Comment by SonixApache** [comment]
¯\_(ツ)_/¯

**Comment by Reviire** [comment]
I'm of the opinion traders should be jobbanned if they don't sell the items and instead horde them/use it for free antag.

**Comment by DamianX** [comment]
>You see heads, engineers, doctors, sec etc get jobbanned all the time for being shitters

**Comment by Accipitermaris** [comment]
@Durgen 
Regarding the Antag Roll thing as a detail: it wasn't explicitly mentioned as a feature in the PR, and I think people are giving Tradercode slack that they wouldn't give if it was regarding lootboxes for antag roles at that point in the round for another other department.

As a thought experiment, imagine if Security started with a random selection of toys to hand out to prisoners for good behavior, but the toys had a 1% chance to contain an item that made you into an Antagonist. Then consider the ensuing discussion surrounding such an emergent feature, considering what would change, and what would remain unchanged.

This isn't a perfect comparison, nor can one exist, as we only have one job that fills the niche of Traders. But hopefully it gets at what I'm talking about.

**Comment by DamianX** [comment]
Leave the masks
Make the masks not work on traders

**Comment by Durgen** [comment]
@DamianX also a good compromise

**Comment by jknpj** [comment]
@Durgen i'm seen a bird roll a stone mask + spider mask in one round, that's for sure

**Comment by Kevak** [comment]
Lets go over each proposed removal.

-I have never rolled stone mask before as a trader. 
-I've rolled warping claws thrice out of a good 100ish rounds with the wardrobe. High value sale in one round, (sold to miner for 2k) one round I was an antag and used it for B&E, and one round I used it to loot space vaults via mesons combo. The loot from which was then sold.
-Holochips are commonish as they have several variants+clothes that come with em pre attached. I usually sell em/use em to find vaults to loot to sell shit from. They're great for locating and capturing asteroid facehuggers for sale. Very profitable and surprisingly rarely ends in a xenomorph outbreak. It's more common for the clown to get aborted instead after opening the welded locker. Chips usually go for 75-150 depending on the job of whom is buying.
-Timeclothes are rare to the point of having not seen em at any point. I have legit seen the Administrative Cybernetic Jumpsuit more often than I have seen timeclothes. (4 out of 100ish.) Timeclothes by proxy are ultimately just cosmetic purchases with armor, relegating them to the 60c+ credit range.

**Comment by Irradiation** [comment]
Personally, I'm not too much against birdniggers being able to RNG get some cool clothes/armor, but getting a Stone Mask is a gigantic no-no. If you're a Xenoarch, you'd need to be extremely lucky to find the mask first (and bother _searching_ for it), which is reasonable, since the map is already rolling for the mask, then you have your own (bad) luck that's "rolling" to find the mask.
Next up: Spessmart. Again, first the map rolls for the Spessmart vault to spawn in the first place, then somebody has to _find_ Spessmart and THEN Spessmart rolls for the stone mask.

Or you can be a birdnigger and only have to "roll" once and have it readily available in your wardrobe at roundstart.

**Comment by Kevak** [comment]
Technically birbs roll several times, as it rolls for 'wardrobe being in the trader vender in the first place' and then it rolls for 'is mask present' for each of the 20 clothes spawn slots.

But yah Stone mask should be axed from the list. Free antag is boring. Work for shit.

**Comment by Kurfursten** [comment]
I really think taking for fun stuff like corgi masks away is just inexplicable antifun. This didn't even have a changelog.

**Comment by Kevak** [comment]
I agree with Kurfursten. I do not approve of this PR.

**Comment by Allspice** [comment]
I don't see why traders shouldn't get to sell corgi masks: After you add a whitelist.

**Pull request merged**